rust/src/file.rs

```diff
@@ -176,11 +175,22 @@ impl File {
             // fill / modify buffer
             buf_mat.copy_from(cur_dst_pos, src_mat, cur_src_box)?;
 
-            // write data
+            if self.header.block_type == BlockType::LZ4 || self.header.block_type == BlockType::LZ4HC {
```

Comment:
I'm nearly 99 % sure that this if-block is not needed :)

The `self.block_idx` variable is set by the call to `Dataset::write_block` a few lines later. So, the if-block only every has an effect (i.e., avoiding a seek) for the very first block. It's true that this statement avoids a (currently) unnecessary seek for the very first block. Still, to reduce complexity, I'd kick it out.

Reply:
You're correct! In my mind, I was still thinking of the "sparse write" case :)